### DISCUSSION TITLE
No Wazuh Agent Logs

### DISCUSSION THREAD

**Discussion by LS3-Allen** [Security-Onion-Solutions/securityonion#5629] [Unsupported Versions] [answered]
v2.3.70
ISO install
Distributed
so-status good

I recently started deploying Wazuh agents across my network.  I have configured the firewall to allow agent communication and registration.  I am able to get the Wazuh agents connected and registered to the Manager.  I have installed the Wazuh agent on 3 Windows machines so far, but I do not see them.  The only agents that I see in Kibana/Hosts are the three nodes in my deployment.

I've checked the elasticsearch, filebeat, and logstash logs, but I'm not seeing any errors.  I feel like I'm missing something simple.  Are there any other logs I should check?

Here is the output of so-wazuh-agent-manage / L:
`[hawking@bg-soman-01 ~]$ sudo so-wazuh-agent-manage
[sudo] password for hawking:

****************************************
* Wazuh v3.13.1 Agent manager.         *
* The following options are available: *
****************************************
   (A)dd an agent (A).
   (E)xtract key for an agent (E).
   (L)ist already added agents (L).
   (R)emove an agent (R).
   (Q)uit.
Choose your action: A,E,L,R or Q: L

Available agents:
   ID: 001, Name: bg-soman-01, IP: 10.2.10.100
   ID: 003, Name: ALLEN-PC-02, IP: any
   ID: 005, Name: DAFFPDC, IP: any
   ID: 006, Name: SPARTAN, IP: any
`
I do not see any of those agents in Hunt or Kibana.  Any help is greatly appreciated!

**Comment by trb143** [comment]
have you configured the wazuh.conf file on each windows machine with the IP address of the master and added the individual keys?

**Reply by LS3-Allen** [reply]
@trb143 
Yes.  Each agent conf file is pointing to the IP address of the manager and I have added the keys.  I looked in /var/ossec/logs/archives/archives.json and it looks like the Wazuh logs are in there from the agents.

**Comment by dougburks** [comment]
In Kibana, have you checked the Wazuh dashboard to see if you can see the logs there?

You can also check our Hunt interface as it has several default queries for Wazuh logs:
https://docs.securityonion.net/en/2.3/hunt.html

**Reply by LS3-Allen** [reply]
I have checked the Kibana Wazuh dashboard and Hunt.  The only logs I see are from the 3 heavy nodes.

I noticed in the Kibana Host dashboard that each heavy node is also listed as a Wazuh manager.  Is that correct?  I thought the Manager node was the Wazuh manager.

![image](https://user-images.githubusercontent.com/84197174/134925431-4c190bf6-5554-4499-90c3-7ba4fd9640d4.png)

**Reply by LS3-Allen** [reply]
I enabled debugging on the Filebeat log and see a lot of entries like this:

```
2021-09-27T21:02:46.117Z        DEBUG   [processors]    processing/processors.go:128    Fail to apply processor client{drop_fields={"Fields":["source","prospector","input","offset","beat"],"IgnoreMissing":false}}: failed to drop field [source]: key not found; failed to drop field [prospector]: key not found; failed to drop field [offset]: key not found; failed to drop field [beat]: key not found
2021-09-27T21:02:46.118Z        DEBUG   [processors]    processing/processors.go:203    Publish event: {
  "@timestamp": "2021-09-27T21:02:46.117Z",
  "@metadata": {
    "beat": "filebeat",
    "type": "_doc",
    "version": "7.13.4",
    "pipeline": "ossec"
  },
  "module": "ossec",
  "category": "host",
  "host": {
    "name": "bg-soman-01"
  },
  "agent": {
    "version": "7.13.4",
    "hostname": "so-filebeat",
    "ephemeral_id": "60aa0c61-2094-44bb-888b-6b6ebe716d8b",
    "id": "9dc21e22-c0b0-4a83-ae04-7f580f83393b",
    "name": "bg-soman-01",
    "type": "filebeat"
  },
  "log": {
    "offset": 191731668,
    "file": {
      "path": "/wazuh/archives/archives.json"
    }
  },
```
That is followed by the Windows Event log captured by the Wazuh agent.  Does the "key not found" message above have something to do with Elasticsearch?

**Reply by dougburks** [reply]
> I noticed in the Kibana Host dashboard that each heavy node is also listed as a Wazuh manager. Is that correct? I thought the Manager node was the Wazuh manager.

Each Security Onion instance runs its own Wazuh manager.

**Reply by LS3-Allen** [reply]
I have triple checked all of my configuration and everything looks good.  I have even redeployed my entire distribution 3 times in the hopes that maybe I missed something.

I checked /nsm/wazuh/archives/archives.json and I am seeing logs from the Wazuh agent in there.  It seems like they just aren't making it to Elasticsearch.  I checked the Filebeat logs just now and I see these errors:
```
2021-09-30T13:30:05.837Z        ERROR   [logstash]      logstash/async.go:280   Failed to publish events caused by: EOF
2021-09-30T13:30:05.838Z        ERROR   [logstash]      logstash/async.go:280   Failed to publish events caused by: client is not connected
2021-09-30T13:30:06.912Z        ERROR   [publisher_pipeline_output]     pipeline/output.go:180  failed to publish events: client is not connected
```
I checked the Logstash logs and there are no new entries in it since 10:41 pm last night.  So-status shows that Logstash is OK.

The only entries in the Elasticsearch logs are in reference to script.max_compilations_rate being deprecated and removed in a future release.

The only thing I have to go off of is the Filebeat log, which points to an issue with Logstash.  Any suggestions on what I should check with Logstash?

**Comment by LS3-Allen** [comment]
Does anyone have any other suggestions that might point me in the right direction to solving this issue?  Am I at least correct in my thinking that Filebeat is not sending logs to Logstash, or am I not even close?

I turned on debug in Filebeat logs and I see these messages:

`2021-10-04T21:30:22.840Z        DEBUG   [processors]    processing/processors.go:128    Fail to apply processor client{drop_fields={"Fields":["source","prospector","input","offset","beat"],"IgnoreMissing":false}}: failed to drop field [source]: key not found; failed to drop field [prospector]: key not found; failed to drop field [offset]: key not found; failed to drop field [beat]: key not found`

**Reply by dougburks** [reply]
Are all other logs coming in normally?

**Reply by LS3-Allen** [reply]
> Are all other logs coming in normally?

Yes, it seems like all other logs are viewable.  What puzzles me even more is I do see the Wazuh logs from each of the heavy nodes.  I just do not see them for any of my Windows machines with the Wazuh agent installed.  I checked /nsm/wazuh/logs/archives/archives.json and I do see Windows Event logs from my agents in there.  Correct me if I'm wrong, but that makes me think the Wazuh agents are connecting and sending logs as they should.

![image](https://user-images.githubusercontent.com/84197174/136219861-cb6aa7a3-9c7f-4d3c-88db-74a30d350c6a.png)

**Reply by dougburks** [reply]
> I checked /nsm/wazuh/logs/archives/archives.json and I do see Windows Event logs from my agents in there. Correct me if I'm wrong, but that makes me think the Wazuh agents are connecting and sending logs as they should.

Yes, it does seem like your Windows agents are sending logs properly and perhaps they're just not generating alerts.

Try removing `AND event.dataset:alert` from the Kibana query and see if you get more logs. 

You could also grab a snippet from one of those Windows logs in archives.json and search Kibana or our Hunt interface.

**Reply by LS3-Allen** [reply]
> > I checked /nsm/wazuh/logs/archives/archives.json and I do see Windows Event logs from my agents in there. Correct me if I'm wrong, but that makes me think the Wazuh agents are connecting and sending logs as they should.
> 
> Yes, it does seem like your Windows agents are sending logs properly and perhaps they're just not generating alerts.
> 
> Try removing `AND event.dataset:alert` from the Kibana query and see if you get more logs.
> 
> You could also grab a snippet from one of those Windows logs in archives.json and search Kibana or our Hunt interface.

I removed 'AND event.dataset:alert' from the Kibana query and it did not show any more logs.

I found Microsoft-Windows-Security-Auditing event logs in archives.json and searched in Hunt for keywords "AUDIT_FAILURE" and also "AUDIT_SUCCESS." from that log.  It did not show any results.

![image](https://user-images.githubusercontent.com/84197174/136396528-340a234d-700b-4424-8b0f-b87ceaa520a7.png)
![image](https://user-images.githubusercontent.com/84197174/136396632-ec9d6de6-0556-40b3-9aeb-6fb0470015b2.png)
![image](https://user-images.githubusercontent.com/84197174/136396705-44289692-a485-49aa-b497-57d38606d959.png)

**Reply by LS3-Allen** [reply]
Here is a sanitized snippet from archives.json.

```
{"timestamp":"2021-10-07T13:28:40.916+0000","rule":{"level":4,"description":"Failed attempt to perform a privileged operation","id":"60107","mitre":{"id":["T1078"],"tactic":["Defense Evasion","Initial Access","Persistence","Privilege Escalation"],"technique":["Valid Accounts"]},"firedtimes":393,"mail":false,"groups":["windows","windows_security","authentication_success"],"pci_dss":["10.2.2"],"gdpr":["IV_32.2"],"hipaa":["164.312.b"],"nist_800_53":["AU.14","AC.6"],"tsc":["CC6.8","CC7.2","CC7.3"]},"agent":{"id":"002","name":"ALLEN-PC-02","ip":"x.x.x.x"},"manager":{"name":"bg-soman-02-wazuh-manager"},"id":"1633613320.1154865","full_log":"{\"win\":{\"system\":{\"providerName\":\"Microsoft-Windows-Security-Auditing\",\"providerGuid\":\"{54849625-5478-4994-a5ba-3e3b0328c30d}\",\"eventID\":\"4673\",\"version\":\"0\",\"level\":\"0\",\"task\":\"13056\",\"opcode\":\"0\",\"keywords\":\"0x8010000000000000\",\"systemTime\":\"2021-10-07T13:28:40.1240913Z\",\"eventRecordID\":\"2480564\",\"processID\":\"4\",\"threadID\":\"14928\",\"channel\":\"Security\",\"computer\":\"ALLEN-PC-02.domain.com\",\"severityValue\":\"AUDIT_FAILURE\",\"message\":\"\\\"A privileged service was called.\\r\\n\\r\\nSubject:\\r\\n\\tSecurity ID:\\t\\tS-1-5-21-823518204-1637723038-1801674531-8998171\\r\\n\\tAccount Name:\\t\\tallen\\r\\n\\tAccount Domain:\\t\\tDOMAIN\\r\\n\\tLogon ID:\\t\\t0x20DD1D8\\r\\n\\r\\nService:\\r\\n\\tServer:\\tSecurity\\r\\n\\tService Name:\\t-\\r\\n\\r\\nProcess:\\r\\n\\tProcess ID:\\t0x610\\r\\n\\tProcess Name:\\tC:\\\\Windows\\\\System32\\\\RuntimeBroker.exe\\r\\n\\r\\nService Request Information:\\r\\n\\tPrivileges:\\t\\tSeLoadDriverPrivilege\\\"\"},\"eventdata\":{\"subjectUserSid\":\"S-1-5-21-823518204-1637723038-1801674531-8998171\",\"subjectUserName\":\"allen\",\"subjectDomainName\":\"DOMAIN\",\"subjectLogonId\":\"0x20dd1d8\",\"objectServer\":\"Security\",\"privilegeList\":\"SeLoadDriverPrivilege\",\"processId\":\"0x610\",\"processName\":\"C:\\\\\\\\Windows\\\\\\\\System32\\\\\\\\RuntimeBroker.exe\"}}}","decoder":{"name":"windows_eventchannel"},"data":{"win":{"system":{"providerName":"Microsoft-Windows-Security-Auditing","providerGuid":"{54849625-5478-4994-a5ba-3e3b0328c30d}","eventID":"4673","version":"0","level":"0","task":"13056","opcode":"0","keywords":"0x8010000000000000","systemTime":"2021-10-07T13:28:40.1240913Z","eventRecordID":"2480564","processID":"4","threadID":"14928","channel":"Security","computer":"ALLEN-PC-02.domain.com","severityValue":"AUDIT_FAILURE","message":"\"A privileged service was called.\r\n\r\nSubject:\r\n\tSecurity ID:\t\tS-1-5-21-823518204-1637723038-1801674531-8998171\r\n\tAccount Name:\t\tallen\r\n\tAccount Domain:\t\tDOMAIN\r\n\tLogon ID:\t\t0x20DD1D8\r\n\r\nService:\r\n\tServer:\tSecurity\r\n\tService Name:\t-\r\n\r\nProcess:\r\n\tProcess ID:\t0x610\r\n\tProcess Name:\tC:\\Windows\\System32\\RuntimeBroker.exe\r\n\r\nService Request Information:\r\n\tPrivileges:\t\tSeLoadDriverPrivilege\""},"eventdata":{"subjectUserSid":"S-1-5-21-823518204-1637723038-1801674531-8998171","subjectUserName":"allen","subjectDomainName":"DOMAIN","subjectLogonId":"0x20dd1d8","objectServer":"Security","privilegeList":"SeLoadDriverPrivilege","processId":"0x610","processName":"C:\\\\Windows\\\\System32\\\\RuntimeBroker.exe"}}},"location":"EventChannel"}
```

**Comment by dougburks** [comment]
Are you able to duplicate this on a fresh installation (perhaps just a VM)?

**Comment by LS3-Allen** [comment, marked as answer]
Update....

I finally got Wazuh alerts showing up in my SOC Hunt and Kibana pages.  My guess is in one of the previous updates changes were made to the architecture.  Previously I had 1 manager and 3 heavynodes running just fine.  My solution was to change the manager to a manager+search node.  Since doing that my Wazuh agent alerts and logs are visible in SOC.

**Reply by dougburks** [reply]
Yes, earlier versions had a bug where heavy nodes were incorrectly pulling logs from the redis queue on the manager. This was corrected in a recent version so that heavy nodes only store their own local logs.
